src/main/drivers/barometer/barometer_lps.c

```diff
@@ -0,0 +1,166 @@
+#define LPS25HB_RPDS_H      0x3A
+//=======================================Constants=============================================// 
+#define LPS25HB_ID          0xBD
+#define LPS25HB_READ        0x80
+#define LPS25HB_MULT        0x40
+//======================================ODR Values=============================================//
+// CTRL1 value
+#define LPS25HB_ODR_ONE_SHOT        0x00
+#define LPS25HB_ODR_1               0x10
+#define LPS25HB_ODR_7               0x20
+#define LPS25HB_ODR_12_5            0x30
+#define LPS25HB_ODR_25              0x40
+//======================================Average number=============================================//
+#define LPS25HB_AVP_8               0x00
+#define LPS25HB_AVP_32              0x01
+#define LPS25HB_AVP_128             0x02
+#define LPS25HB_AVP_512             0x03
+#define LPS25HB_AVT_8               0x00
+#define LPS25HB_AVT_16              0x04
+#define LPS25HB_AVT_32              0x08
+#define LPS25HB_AVT_64              0x0C
+//======================================Moving average number=============================================//
+#define LPS25HB_NO_MA               0x00
+#define LPS25HB_MA_2                0x01
+#define LPS25HB_MA_4                0x03
+#define LPS25HB_MA_8                0x07
+#define LPS25HB_MA_16               0x0F
+#define LPS25HB_MA_32               0x1F
+
+bool lpsWriteCommand(busDevice_t *busdev, uint8_t cmd, uint8_t byte) {
+    return spiBusWriteRegister(busdev, cmd, byte);
+}
+
+bool lpsReadCommand(busDevice_t *busdev, uint8_t cmd, uint8_t len, uint8_t *data) {
+    return spiBusReadRegisterBuffer(busdev, cmd | 0x80 | 0x40, data, len);
+}
+
+static void lpsOn(busDevice_t *busdev, uint8_t CTRL1_val) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, CTRL1_val | 0x80);
+    delay(500);
+}
+
+static void lpsOff(busDevice_t *busdev) {
+    lpsWriteCommand(busdev, LPS25HB_CTRL1, 0x00 | (0x01 << 2));
+}
+
+uint32_t rawP = 0;
+uint32_t rawT = 0;
+
+static void lps_nothing(baroDev_t *baro) {
+	UNUSED(baro);
+	return;
+}
+
+static void lps_read(baroDev_t *baro) {
+    uint8_t status = 0x00;
+    lpsReadCommand(&baro->busdev, LPS25HB_STATUS, 1, &status);
+    if (status & 0x03) {
+    		uint8_t temp[5];
+    		lpsReadCommand(&baro->busdev, LPS25HB_OUT_XL, 5, temp);
+
+    		/* Build the raw data */
+    		for(uint8_t i=0; i<3; i++)
+    			rawP |= (((uint32_t)temp[i]) << (8*i));
+    		if (rawP & 0x00800000)
+    		rawP |= 0xFF000000;
+
+    		//Temp in C
+    		uint16_t temperature_tmp = ((uint16_t) temp[4] << 8) | (uint16_t) temp[3];
+    		rawT = (((uint16_t)temperature_tmp / 48 + 425) * 10);
+    } else {
+    		rawP = 0;
+    		rawT = 0;
+    }
+}
+
+static void lps_calculate(int32_t *pressure, int32_t *temperature) {
+	*pressure = (int32_t)((float)((int32_t)rawP) * 6.25f); //should be divided by 40.96f
+	*temperature = (int32_t)rawT;
+}
+
+bool lpsDetect(baroDev_t *baro)
+{
+
+    delay(10); // No idea how long the chip takes to power-up, but let's make it 10ms
```

Comment:
No delay is probably needed (Assuming that datasheet would mention it if it is important)